src/frontend/next/src/pages/_document.tsx

```diff
@@ -37,12 +36,20 @@ MyDocument.getInitialProps = async (ctx) => {
     });
 
   const initialProps = await Document.getInitialProps(ctx);
+  const styles = extractCritical(initialProps.html);
 
   return {
     ...initialProps,
     // Styles fragment is rendered after the app and page rendering finish.
-    styles: [...Children.toArray(initialProps.styles), sheets.getStyleElement()],
+    styles: [
+      ...Children.toArray(initialProps.styles),
+      sheets.getStyleElement(),
+      <style
+        key="emotion-style-tag"
+        data-emotion={`css ${styles.ids.join(' ')}`}
+        // eslint-disable-next-line react/no-danger
```

Comment:
We can do this in the global eslint config, since we'll need it for post content too.